src/classes/TestFactory.cls

```diff
@@ -146,4 +203,130 @@ public class TestFactory {
 			};
 		}
 	}
-}
+
+	/**
+	 * The method automatically setups random values for required custom fields for the given sobject.
```

Comment:
When building this test factory, I consciously did not want to automate any random data generation. There was a popular project at https://github.com/mbotos/SmartFactory-for-Force.com, and I ran into problems with validation rules and other things when using it. I prefer to make sure as a developer I understand the schema and data types that need to go into the test data. This is probably more philosophical, but I just like knowing what exactly the shape is of my test data.
  
I wonder if there's be another way to approach this to allow the test factory to be extended for people to add auto-generation, but not have it included as part of the base code.

Reply:
You are right, developers should know the data. I just wanted to provide the option, because in some specific cases, it's useful. For example, I am currently working on a global implementation, and we have issues because a lot of people are admin, so you would see required fields being added very often for instance. Tests break often and slowing us down. Auto generation for data coming from other streams (and that we don't want to test anyway) could be useful. To use with caution of course.
Anyway, I get your point and I will keep my work on the fork for now, as an alternative to your TestFactory.
Thank you for you time providing the feedback!